docs/configuration-reference/backend/backend.md

```diff
@@ -0,0 +1,78 @@
+# Backend configuration reference
+
+## Contents
+
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Configuration](#configuration)
+* [Argument reference](#argument-reference)
+
+## Introduction
+
+Lokomotive supports local or remote backend [S3 only] for storing Terraform state.
+Lokomotive also supports optional state locking feature for S3 backend.
+
+Backend configuration is **OPTIONAL**. If no backend configuration is provided then local backend is
+used.
+
+>NOTE: Installer does not support multiple backends, configure only one.
+
+## Prerequisites
+
+There are no prerequisites for using local backend.
+
+The following is required for configuring AWS S3 backend:
+
+* AWS S3 bucket to be used should already be created.
+* DynamoDB table to be used for state locking should already be created.
+* Correct IAM permissions for the S3 bucket and DynamoDB Table. At minimum the following are the
+  permissions required:
+  * [S3 bucket permissions](https://www.terraform.io/docs/backends/types/s3.html#s3-bucket-permissions).
+  * [DynamoDB table permissions](https://www.terraform.io/docs/backends/types/s3.html#dynamodb-table-permissions).
+
+## Configuration
+
+To use a backend, we need to define a configuration in the `.lokocfg` file.
+
+Example configuration file:
+
+### Local backend
+
+```hcl
+backend "local" {
```

Comment:
Actually, maybe both backends should have separate file for documentation?

Reply:
made changes regarding this.

Reply:
@invidian separated into local.md and s3.md